updater/lib/dependabot/api_client.rb

```diff
@@ -18,10 +19,11 @@
 module Dependabot
   class ApiError < StandardError; end
 
-  class ApiClient
+  class ApiClient # rubocop:disable Metrics/ClassLength
     extend T::Sig
 
     MAX_REQUEST_RETRIES = 3
+    INVALID_REQUEST_MSG = /The request contains invalid or unauthorized changes/
```

Comment:
do you know why we do this?

Reply:
To quote the API code, it's because we are trying to update GitHub Actions workflow files when our package manager is not github_actions or the dependency file is not one of the files specified in the `allowed list` .

It was added as a fix to prevent any potential security issues raised in (issue)[
 https://github.com/github/dependabot-updates/issues/5840#issuecomment-2187039582]